clusterloader2/pkg/flags/types.go

```diff
@@ -57,6 +58,36 @@ func (s *stringFlagFunc) Type() string {
 	return "string"
 }
 
+type stringSliceFlagFunc struct {
+	valPtr         *[]string
+	initializeFunc func() error
+}
+
+// initialize runs additional parsing function.
+func (s *stringSliceFlagFunc) initialize() error {
+	return s.initializeFunc()
+}
+
+// String returns default string.
+func (*stringSliceFlagFunc) String() string {
+	return "false"
+}
+
+// Set handles flag value setting.
+func (s *stringSliceFlagFunc) Set(val string) error {
+	if val != "" {
+		*s.valPtr = strings.Split(val, ",")
+	} else {
+		*s.valPtr = nil
```

Comment:
Do we need this `else`? Isn't `nil` a default value?

Reply:
The flag supports setting both via env var and via flag. So let's say it's MY_STRING_LIST env or --my_string_list flag.

Let's say the MY_STRING_LIST is set to "1,2,3" in some upstream code that we cannot really change, but we want to set it to empty list via flag override. This else clause allows it, I can empty the list by setting --my_string_list=""

Reply:
Ok. :+1: